instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java

```diff
@@ -65,8 +66,9 @@ public static ConfigProperties create(
   @Nullable
   @Override
   public String getString(String name) {
-    String value = environment.getProperty(name, String.class);
-    if (value == null && name.equals("otel.exporter.otlp.protocol")) {
+    String normalizedName = ConfigUtil.normalizePropertyKey(name);
+    String value = environment.getProperty(normalizedName, String.class);
+    if (value == null && normalizedName.equals("otel.exporter.otlp.protocol")) {
```

Comment:
```suggestion
    if (value == null && "otel.exporter.otlp.protocol".equals(normalizedName) {
```
Maybe this is better, it can avoid the influence from upstream API. The line 116 is same.

Reply:
From my understanding, `normalizedName` can't be `null` at this point. 

I would prefer to limit the changes in this PR.

I personally agree with your suggestion.

Perhaps this change could be discussed in another PR. Feel free to create it.